src/matlab2tikz.m

```diff
@@ -1000,12 +1034,51 @@ case guitypes()
 end
 end
 % ==============================================================================
+function m2t = getPlotyyReferences(m2t,axisHandle)
+
+plotyyPeer = getappdata(axisHandle,'graphicsPlotyyPeer');
+% Not a plotyy plot
+if isempty(plotyyPeer)
+    m2t.axesContainers{end}.PlotyyAxisType = '';
+    return
+end
+
+legendHandle = m2t.axesContainers{end}.LegendHandle;
+
+% Main plotyy axis
+if ~isempty(legendHandle)
+    m2t.axesContainers{end}.PlotyyAxisType = 'main';
+    
+    % Create plotyy references to legend entries of the main axis 
+    legendEntries = m2t.axesContainers{end}.LegendEntries;
+    ancAxes       = ancestor(legendEntries,'axes');
+    idx           = ismember(double([ancAxes{:}]), axisHandle);
+    m2t.axesContainers{end}.PlotyyReferences = legendEntries(idx);
+    
+    % Do not create a legend on the main axis
+    m2t.axesContainers{end}.LegendHandle = [];
+else
+    m2t.axesContainers{end}.PlotyyAxisType = 'secondary';
+    
+    % Get legend handle plotyy main axis
+    m2t.axesContainers{end}.LegendHandle = getAssociatedLegend(m2t,plotyyPeer);
+    
+    % Get legend entries associated to secondary plotyy axis
+    legendEntries = getLegendEntries(m2t);
+    ancAxes       = ancestor(legendEntries,'axes');
+    idx           = ismember(double([ancAxes{:}]), axisHandle);
+    m2t.axesContainers{end}.LegendEntries = legendEntries(idx);
+    
+    % Reference the main axis legend entries
+    m2t.axesContainers{end}.PlotyyReferences = legendEntries(~idx);
+end
+
+end
+% ==============================================================================
 function string = getLegendString(m2t, h)
 % Retrieve the legend string for the given handle 
 
-entries = m2t.axesContainers{end}.LegendEntries;
-idx     = ismember(entries, h);
-string  = getOrDefault(entries(idx), 'displayname', '');
+string  = getOrDefault(h, 'displayname', '');
```

Comment:
No need to compare against legend entries since I need to call it on `PlotyyReferences` too. If handle does not have legend string, it will be empty.